emu/src/cpu/arm7tdmi.rs

```diff
@@ -1023,18 +1057,50 @@ impl From<u8> for HalfwordTransferType {
 #[cfg(test)]
 mod tests {
     use crate::cpu::condition::Condition;
+    use crate::cpu::instruction::ArmModeInstruction::{Branch, BranchAndExchange};
     use pretty_assertions::assert_eq;
 
     use super::*;
 
     #[test]
-    fn test_branch() {
+    fn check_branch_and_exchange() {
+        {
+            let op_code = 0b1110_0_0_0_1_0_0_1_0_1_1_1_1_1_1_1_1_1_1_1_1_0_0_0_1_0000;
+            let cpu = Arm7tdmi::default();
+            let op_code: ArmModeOpcode = cpu.decode(op_code);
+            assert_eq!(op_code.instruction, BranchAndExchange(Condition::AL, 0));
+            let asm = op_code.instruction.disassembler();
+            assert_eq!(asm, "BX R0");
+        }
```

Comment:
Here we're not really testing for the functionality, right?

Reply:
Nope but is an additional test that before didn't exist :)

Reply:
Sure but maybe we could add also the check for the pc and the status (arm/thumb) after the execution, if we don't have this test, once we're already touching it